csrc/host_ir/lower_to_communication.cpp

```diff
@@ -386,10 +397,12 @@ std::optional<CommunicationInfo> getCommunicationInfo(Expr* expr) {
           CommunicationType::ReduceScatter,
           c2p_map.at(c_logical_id),
           c_logical_id);
-    } else {
-      NVF_THROW("Unsupported expression: ", expr->toString());
     }
   }
+
+  if (!communication_info.has_value()) {
```

Comment:
This implies that the expression must be resharding, so if the above logic has not found a comm yet, it must be broadcast.
We should assert on the expression being `isResharding` at the beginning of the code.

Reply:
I see this call is being called after `isResharding` in the two preseg passes. So the comment maybe sufficient to avoid calling the expensive `isResharding` twice.

Ideally, I do prefer the check being here so I don't have to remember it.